### ISSUE TITLE
Unicode compatibility

### ISSUE THREAD

**Issue by musically-ut** [miracle2k/python-glob2#5]
I have had [some issues](http://stackoverflow.com/questions/29903338/filesystem-independent-way-of-using-glob-glob-and-regular-expressions-with-unico) in globbing and matching filenames with unicode characters on different OSes/filesystems.

@remram44 has suggested [a solution](http://stackoverflow.com/a/29972288/987185) which I think will be a good addition to `glob2`. The only issue I can see is that it may be slower than the current code though I haven't benchmarked it.

If you think it would be useful, I can work on a pull request with this feature.

**Comment by miracle2k** [comment]
Yes, I'm open to that. Maybe the Globber class could have an option to turn the behaviour on/off.
